auth/src/test/kotlin/com/hedvig/android/auth/AccessTokenAuthenticatorTest.kt

```diff
@@ -125,28 +75,39 @@ class AccessTokenAuthenticatorTest {
     assertThat(firstRequest.headers["Authorization"]).isEqualTo(expiredToken)
     // Second request intercepted by authenticator, adding an updated token
     val secondRequest = mockWebServer.takeRequest()
-    assertThat(secondRequest.headers["Authorization"]).isEqualTo("newToken")
+    assertThat(secondRequest.headers["Authorization"]).isEqualTo(refreshedAccessToken.token)
   }
 
   @Test
-  fun shouldPersistNewTokenWhenResponseIs401AndCallIsSuccessFull() {
+  fun shouldPersistNewTokenWhenResponseIs401AndCallIsSuccessful() = runTest(UnconfinedTestDispatcher()) {
```

Comment:
These tests rely on using the `Authenticator` we have. But that one uses `runBlocking` to go from non-coroutines into coroutines world. This means that we don't quite have the same control over it to pass the test dispatcher and be able to do `runCurrent()` on it, so we're using the UnconfinedTestDispatcher to simply have everything run eagerly and it works.